filesystem/iso9660.ksy

```diff
@@ -216,60 +94,846 @@ types:
+                            type: u1
+                          - id: version
+                            contents: [ 0x1 ]
+                          - id: file_mode
+                            type: u4bi
+                          - id: links
+                            type: u4bi
+                          - id: user
+                            type: u4bi
+                          - id: group
+                            type: u4bi
+                          - id: serial
+                            type: u4bi
+                            if: length >= 44
+                      rrip_re:
+                        doc-ref: rrip 4.1.5.3
+                        seq:
+                          - id: length
+                            type: u1
+                          - id: version
+                            contents: [ 0x1 ]
+                      rrip_sf:
+                        doc-ref: rrip 4.1.7
+                        seq:
+                          - id: length
+                            type: u1
+                          - id: version
+                            contents: [ 0x1 ]
+                          - id: virtual_file_size_high
+                            type: u4bi
+                          - id: virtual_file_size_low
+                            type: u4bi
+                          - id: table_debth
+                            type: u1
+                            enum: table_debth
+                        enums:
+                          table_debth:
+                            0x2: max_64kb
+                            0x4: max_64mb
+                            0x8: max_4gb
+                            0x10: max_1tb
+                            0x20: max_256tb
+                            0x40: max_64pb
+                            0x80: max_16eb
+                      susp_sl:
+                        doc-ref: susp 4.1.3.1
+                        seq:
+                          - id: length
+                            type: u1
+                          - id: version
+                            contents: [ 0x1 ]
+                          - id: reserved
+                            type: b4
+                          - id: root
+                            type: b1
+                          - id: parent
+                            type: b1
+                          - id: current
+                            type: b1
+                          - id: continue
+                            type: b1
+                          - id: content
+                            type: str
+                            size: length - 5
+                      susp_sp:
+                        doc-ref: susp 5.3
+                        seq:
+                          - id: length
+                            type: u1
+                          - id: version
+                            contents: [ 0x1 ]
+                          - id: check_bytes
+                            contents: [ 0xbe, 0xef ]
```

Comment:
```yaml
type: u2
valid: 0xbeef
```

Reply:
Just to be sure... would both little and big endian do the same with 0xbeef?
As the iso9660 is a big mix of both

Reply:
> Just to be sure... would both little and big endian do the same with 0xbeef?

No, `u2` will be affected by the default endianness. I would leave `contents` here.

Reply:
No, `u2` should use the current endianness (endianness can be switcheable (automatically making all the numbers specified without explicit endianness being of endianness chosen in runtime), though this feature seems to be got broken somewhen and though in this spec it is not used). If one wants it always be `be` for some `u2` field, he can always write `u2be`. Fixed the proposed change to make it correspond to `contents: [0xbe, 0xef]` and make it guaranteedly be `be` even if you would decide to use switcheable endianness.

Reply:
>Also it's literally called check_bytes, so there's no real reason to convert it into an integer (maybe except saving a few CPU clock ticks).

It is 2 bytes, the `be` endianness is used extensively in this spec, CPUs think in words, and 0x(dead)beef is a well-known hex-speak constant ... I feel like it was meant to be used as `u2` by the standards authors.

Reply:
The kind of strange logic in this format is that many times both little and big endian are stored in different fields.
But I did not read all those specs in 2 years...

Reply:
"(I can see 18 occurences of contents: [ 0x1 ] in the file)"
I replaced all 1 byte contents

Reply:
I think the logic of 0xbe 0xef is, you read it as a string and you also reference it as a u2, then the can see if your machine is le or be

Reply:
Such things are usually done in headers when the field is at well-known offset, not in deeply buried types. The only reason I can imagine it be done in a deeply buried type is supporting multiple endiannesses and allowing each piece to have an own endianness, shifting the burden of byte-swapping to the one reading it. What does the standard say?

Reply:
I did document the spec reference so nicely... 
doc-ref: susp 5.3 
susp = http://www.ymi.com/ymi/sites/default/files/pdf/Systems%20Use%20P1281.pdf

And now www.ymi.com is down

Reply:
Yes, I can confirm that, I found the same logic in the Android source code (https://android.googlesource.com/platform/external/syslinux/+/refs/heads/oreo-m4-s3-release/core/fs/iso9660/susp_rr.c#299):

```c
    if (sua[0] != 'S' || sua[1] != 'P' || sua[2] != 7 || sua[3] != 1 ||
	sua[4] != 0xbe || sua[5] != 0xef)
	return 0;
```

So no endianness stuff here.

> And now [www.ymi.com](http://www.ymi.com) is down...

Please update the `doc-ref` links that no longer work with the Wayback Machine mirror, so https://web.archive.org/web/20170404132301/http://www.ymi.com/ymi/sites/default/files/pdf/Systems%20Use%20P1281.pdf in this case.